File: src/GLFMmatlab/mapping_functions/f_n.m
function x = f_n(y, mu, w)
    % transformation function for count data
    % Y -> X (from pseudo-obversations to data)
    if (w == 0)
        error('scaling factor should never be 0');
    end
    x = floor( f_p(y, mu, w) );
    %if sum(x == 0)>0 % TODO: verify if that would be a problem % DONE, OK
    %    warning('when transforming y to x, x has been rounded to zero (count data)');
    %end
end

File: src/GLFMR/f_n.R
#' @description  Transformation function for count data
#' @param y: N-sized vector, where N is the number of observations 
#' @param mu: mean to shift data
#' @param w: weights
#' @Return x transformed data


f_n<-function(y,mu,w){
  if (w == 0){
    stop('scaling factor should never be 0')
  }
  else{
    x <- floor(log( exp(y) + 1 )/w + mu )
  }
    return(x)
}


